conn.c

```diff
@@ -138,16 +138,6 @@ conntickat(Conn *c)
 }
 
 
-// TODO: remove this function by inlining its content into 3 callees places.
-// Reason: conn.c does not use rw anywhere in this file.
-void
-connwant(Conn *c, int rw)
```

Comment:
I was trying to factor some things out of prot.c so the struct definition `struct Conn { ... }` could be moved to conn.c (and `typedef struct Conn Conn` would of course stay in dat.h). That's why this function exists, it was an early step in that refactoring. But of course I didn't get far enough to actually move the struct definition.

I was even hoping to do that for most of the struct definitions some day.

Reply:
Aha, this is why so many `conn*` functions, and some of them are somewhat forced... This does not contradict to my intentions. I have isolated operations on `epollq` (former `dirty`) into functions that can/should be moved into `conn.c`. The same is true about `epollq` that can be hidden in `conn.c` I want to break some global dependencies, but it's hard when everything depends on everything. My decision about this might be not ideal and I can change it later but for now it looks reasonable.

Reply:
Q about moving struct def of `Conn` out of `data.h`? Do you think it's possible because clients would still have to see `*Conn` to be able to interface with functions in `conn.c`?

Reply:
Yeah, I agree you should approach refactoring in whatever way best fits your longer term goals.

It's possible to move the struct declaration to conn.c if client functions use only `*Conn` and never `Conn`. Here's a small example:

ex.h:
```c
typedef struct Blob Blob;
Blob *blob_new();
void blob_test(Blob *b);
```

blob.c:
```c
#include <stdlib.h>
#include <stdio.h>
#include "ex.h"

struct Blob {
    int n;
};

Blob *
blob_new()
{
	Blob *b = malloc(sizeof(Blob));
	b->n = 2;
	return b;
}

void
blob_test(Blob *b)
{
	printf("%d\n", b->n);
}
```

user.c:
```c
#include "ex.h"

int
main()
{
	Blob *b = blob_new();
	blob_test(b);
}
```

You can compile this with `cc *.c` and running `./a.out` should print `2`.

But things like like `malloc(sizeof(Blob))` in user.c won't work.

Reply:
Thanks, it looks like data encapsulation to me. I never knew about this in C.

Reply:
@kr I have a strange question about former "dirty" list. Why do we have that in a first place? Why cannot we call sockwant() exactly in place without putting the conn to delayed list "dirty" first? I cannot find explanation for this, understanding this mechanism feels important.